_sane.c

```diff
@@ -578,7 +579,7 @@ SaneDev_snap(SaneDevObject *self, PyObject *args)
   
       if(progress && progress != Py_None)
         {
-          PyObject *progArgs = Py_BuildValue("ii", imgBufCurLine, imgPrioriLines);
+          PyObject *progArgs = Py_BuildValue("sii", self->name, imgBufCurLine, imgPrioriLines);
```

Comment:
To avoid breaking existing uses, it would be better to pass it as last parameter